gcc/rust/ast/rust-ast-collector.h

```diff
@@ -47,15 +47,36 @@ class TokenCollector : public ASTVisitor
    */
   template <typename T>
   void visit_items_joined_by_separator (T &collection,
-					TokenId separator = COMMA,
-					size_t start_offset = 0,
-					size_t end_offset = 0);
```

Comment:
You need to keep default arguments, those are used in the visitor in many situations, removing them means the code won't compile correctly.